zone/raids.cpp

```diff
@@ -2203,3 +2257,734 @@ void Raid::SetNewRaidLeader(uint32 i)
+					delegatee,
+					slot + 1,
+					results.ErrorMessage().c_str()
+				);
+			}
+		}
+	}
+	das->DelegateAbility = RaidDelegateMainMarker;
+	das->MemberNumber = 0;
+	das->EntityID = c->GetID();
+	strcpy(das->Name, delegatee);
+	QueuePacket(outapp);
+	safe_delete(outapp);
+}
+
+int Raid::FindNextRaidDelegateSlot(int option) 
+{
+	//option 1 FindNextRaidMainMarkerSlot
+	//option 2 FindNextRaidMainAssisterSlot
+	if (option == 1) {
+		for (int i = 0; i < MAX_NO_RAID_MAIN_MARKERS; i++) {
+			if (strlen(main_marker_pcs[i]) == 0) {
+					return i;
+			}
+		}
+		return -1;		
+	}
+	else if (option == 2) {
+		for (int i = 0; i < MAX_NO_RAID_MAIN_ASSISTERS; i++) {
+			if (strlen(main_assister_pcs[i]) == 0) {
+				return i;
+			}
+		}
+		return -1;		
+	}
+}
+
+void Raid::UpdateXTargetType(XTargetType Type, Mob* m, const char* Name)
+{
+	for (const auto& rm : members) {
+		if (!rm.member || rm.is_bot || !rm.member->XTargettingAvailable())
+			continue;
+
+		for (int i = 0; i < rm.member->GetMaxXTargets(); ++i)
+		{
+			if (rm.member->XTargets[i].Type == Type)
+			{
+				if (m) {
+					rm.member->XTargets[i].ID = m->GetID();
+				}
+				else {
+					rm.member->XTargets[i].ID = 0;
+				}
+
+				if (Name) {
+					strncpy(rm.member->XTargets[i].Name, Name, 64);
+				}
+
+				rm.member->SendXTargetPacket(i, m);
+			}
+		}
+	}
+}
+
+void Raid::RaidMarkNPC(Mob* mob, uint32 parameter)
+{
+	Client* c = mob->CastToClient();
+	if (!c || !c->GetTarget() || parameter < 1 || parameter > 3) {
+		LogDebug("RaidMarkNPC Failed sanity checks.");
+		return;
+	}
+
+	for (int i = 0; i < MAX_NO_RAID_MAIN_MARKERS; i++) {
+		auto cname = c->GetCleanName();
+		if (strcasecmp(main_marker_pcs[i], cname) == 0 || strcasecmp(leadername, cname) == 0) {
+			marked_npcs[parameter - 1] = c->GetTarget()->GetID();
+
+			std::string query = StringFormat("UPDATE raid_details SET marked_npc%i = %i WHERE raidid = %i;",
+				parameter,
+				marked_npcs[parameter - 1],
+				this->GetID()
+			);
+			auto results = database.QueryDatabase(query);
+			if (!results.Success()) {
+				LogError("Unable to set MarkedNPC{} from slot: [{}] with error [{}]\n",
+					parameter,
+					parameter - 1,
+					results.ErrorMessage().c_str()
+				);
+			}
+
+			auto outapp = new EQApplicationPacket(OP_MarkRaidNPC, sizeof(MarkNPC_Struct));
+			MarkNPC_Struct* mnpcs = (MarkNPC_Struct*)outapp->pBuffer;
+			mnpcs->TargetID = marked_npcs[parameter - 1];
+			mnpcs->Number = parameter;
+			strcpy(mnpcs->Name, c->GetTarget()->GetCleanName());
+			QueuePacket(outapp);
+			safe_delete(outapp);
+			UpdateXtargetMarkedNPC();
+			return;
+		}
+	}
+		//client is not delegated the mark ability
+		c->MessageString(Chat::Cyan, NOT_DELEGATED_MARKER);
+		return;
+}
+
+void Raid::UpdateXtargetMarkedNPC() 
+{
+	for (int i = 0; i < MAX_MARKED_NPCS; i++) {
+		auto mm = entity_list.GetNPCByID(marked_npcs[i]);
+		if (mm) {
+			UpdateXTargetType(static_cast<XTargetType>(RaidMarkTarget1 + i), mm->CastToMob(), mm->CastToMob()->GetName());
```

Comment:
Is the + i intentional here? Just making sure it looks right to you

Reply:
Aye, same as previous.  This is on purpose.